src/inmanta/server/validate_filter.py

```diff
@@ -183,3 +185,64 @@ def process_filters(self, filter: Dict[str, List[str]]) -> Dict[str, QueryFilter
             query["action"] = (QueryType.CONTAINS, validated_filter.action)
 
         return query
+
+
+def parse_single(v: object, property_name: str) -> object:
+    """
+    Transform list values to their single element value.
+    """
+    if isinstance(v, list):
+        return more_itertools.one(
+            v,
+            too_short=ValueError(f"Empty '{property_name}' filter provided"),
+            too_long=ValueError(f"Multiple values provided for '{property_name}' filter: {v}"),
+        )
+
+    return v
+
+
+class CompileReportFilterModel(FilterModel):
+    requested: Optional[DateRangeConstraint]
+    started: Optional[bool]
+    completed: Optional[bool]
+    success: Optional[bool]
+
+    @validator("started", pre=True)
+    @classmethod
+    def _started_single(cls, v: object) -> object:
+        return parse_single(v, "started")
+
+    @validator("completed", pre=True)
+    @classmethod
+    def _completed_single(cls, v: object) -> object:
+        return parse_single(v, "completed")
+
+    @validator("success", pre=True)
+    @classmethod
+    def _success_single(cls, v: object) -> object:
+        return parse_single(v, "success")
+
+    @validator("requested", pre=True)
+    @classmethod
+    def parse_requested(cls, v: object) -> List[Tuple[RangeOperator, datetime.datetime]]:
+        return parse_range_operator(v)
+
+
+class CompileReportFilterValidator(FilterValidator[CompileReportFilterModel]):
+    @property
+    def model(self) -> Type[CompileReportFilterModel]:
+        return CompileReportFilterModel
+
+    def process_filters(self, filter: Dict[str, List[str]]) -> Dict[str, QueryFilter]:
+        validated_filter: CompileReportFilterModel = self.validate_filters(filter)
+        query: Dict[str, QueryFilter] = {}
+        if validated_filter.requested:
+            query["requested"] = (QueryType.RANGE, validated_filter.requested)
+        if validated_filter.success is not None:
+            query["success"] = (QueryType.EQUALS, validated_filter.success)
+        if validated_filter.started is not None:
+            query["started"] = (QueryType.IS_NOT_NULL, None) if validated_filter.started else (QueryType.EQUALS, None)
+        if validated_filter.completed is not None:
+            query["completed"] = (QueryType.IS_NOT_NULL, None) if validated_filter.completed else (QueryType.EQUALS, None)
```

Comment:
Why use `IS_NOT_NULL` here? This doesn't match with the API documentation.

Reply:
Could you elaborate? 
For the `started` and `completed` values we only need the `IS_NOT_NULL` filter, as the ticket described it, and I don't think the API docs suggest otherwise

Reply:
You are right. You can ignore this comment.